### ISSUE TITLE
ibm_db2 documentation is inconstant and incomplete

### ISSUE THREAD

**Issue by shemau** [IBM-Cloud/terraform-provider-ibm#5826]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s) or Datasource(s)

* ibm_db2

### Description

This is a provider issue and unrelated to terraform version. 1.5.7, 1.6.x and 1.9.2 are all affected.
This feature was added in 1.72.0-beta0 and this is the only current version of the provider impacted.

ibm_db2_instance = ibm_db2: It is standard for the left menu resource names to match the resource names. The resource/data source are both called ibm_db2 (not ibm_db2_instance).

Argument references (inputs)

- [plan](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/db2_instance#plan-4): the only option is performance , even the example uses performance. The implementation is that the options are free, standard or enterprise. The choice of plan impacts the available locations, eg. free does not appear to run in us-east.

- [backup_location](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/db2_instance#backup_location-5): is the only mention of cross regional versus regional, but it is likely intended to relate to business continuity and disaster recovery in the event of a total MZR failure being better if the backup is in a different MZR (aka location).

- [instance_type](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/db2_instance#instance_type-6) could be clearer. It implies that shared hosts are available, but it seems only isolated (no other tenants) is really available.

Attribute references (outputs)

- [version](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/db2_instance#version-8): is an attribute reference, but there is no sign of in the outputs.

Outputs

- [attribute references](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/db2_instance#attribute-reference): with around 15 arguments and 3 attributes, there should be around 18 total. There are over 50.
Some of these references present real problems in declarative model. 
  - For example, updated_at is a time, every time it changes the consumer will have drift and want to run an apply. 
  - state, status, resource_status apart from being the same thing may bounce between different value or flip/flop. Every time it changes the consumer has drift. 
  - The *_url appear valid.  
  
In summary a lot of these attributes are internal and should not be output, others are value add and should be documented.

# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.

--->
### References

* #0000
